### PR TITLE
[Keyboard] Add Creator Pro by SergioPoverony

### PR DESCRIPTION
**Pull request by sergiopoverony**
## Description

This is keyboard for support making video content. 
Have 8 button and 1 encoder with one button? summary 9 button and Encoder
via layout prepared, but need pull keyboard here 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x]  Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Can you explain why you're using 0x20A0 as the VID, which is currently assigned to Clay Logic?

**Comment by sergiopoverony** [comment]
> Can you explain why you're using 0x20A0 as the VID, which is currently assigned to Clay Logic?

Yes, mistake vusb vid pid cant use in usb. Changed all

File: keyboards/sergiopoverony/creator_pro/config.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    SergioPoverony
+#define PRODUCT         Creator Pro
+
+
+/* Optimise size swith off unused functions*/
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define NO_ACTION_TAPPING
+#define NO_ACTION_ONESHOT
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/config.h:32]: Not a fan of these being defined, as they disable a lot of functionality

File: keyboards/sergiopoverony/creator_pro/keymaps/via/creator_pro.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/keymaps/via/creator_pro.json:1]: This file doesn't belong in this repo.  It needs to be removed and PR'ed to the `the-via/keyboards` repo, instead.

File: keyboards/sergiopoverony/creator_pro/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+This Keypad has 1 rows x 9 columns of keys. It has the top/default layer that
+has a few 1-8 keys as well as one dedicated encoder key for change layer. The
+ second layer has QWERASDF keys. Encoder have 4 function on any layer is 
+ Volume Control, Mouse Wheel, PageUp/PageDn, Left/Righ buttons.
+The keypad also includes three RGB LEDs that show active layer.
+
+Keyboard Maintainer: [SergioPoverony](https://github.com/sergiopoverony)  
+
+Hardware:
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/readme.md:13]: Could you add instructions on how to enter the bootloader to flash new firmware?

File: keyboards/sergiopoverony/creator_pro/rules.mk [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sergiopoverony/creator_pro/config.h (hunk 1 above)
Comment (sergiopoverony) [reply keyboards/sergiopoverony/creator_pro/config.h]: fixed.

File: keyboards/sergiopoverony/creator_pro/keymaps/via/creator_pro.json (hunk 2 above)
Comment (sergiopoverony) [reply keyboards/sergiopoverony/creator_pro/keymaps/via/creator_pro.json]: Yes. i remove this.

File: keyboards/sergiopoverony/creator_pro/readme.md (hunk 3 above)
Comment (sergiopoverony) [reply keyboards/sergiopoverony/creator_pro/readme.md:13]: Add in Hardware section

File: keyboards/sergiopoverony/creator_pro/rules.mk (hunk 4 above)
Comment (sergiopoverony) [reply keyboards/sergiopoverony/creator_pro/rules.mk]: removerd. Thank you

File: keyboards/sergiopoverony/creator_pro/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ #include QMK_KEYBOARD_H
+
+/* define layers */
+#define _HOME 0
+#define _RED 1
+#define _BLUE 2
+#define _GREEN 3
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/keymaps/default/keymap.c:23]:
```suggestion
enum layers {
    _HOME,
    _RED,
    _BLUE,
    _GREEN,
);
```

File: keyboards/sergiopoverony/creator_pro/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ #include QMK_KEYBOARD_H
+
+/* define layers */
+#define _HOME 0
+#define _RED 1
+#define _BLUE 2
+#define _GREEN 3
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/keymaps/via/keymap.c:23]:
```suggestion
enum layers {
    _HOME,
    _RED,
    _BLUE,
    _GREEN,
);
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sergiopoverony/creator_pro/keymaps/default/keymap.c (hunk 5 above)
Comment (sergiopoverony) [reply keyboards/sergiopoverony/creator_pro/keymaps/default/keymap.c]: enumed

File: keyboards/sergiopoverony/creator_pro/keymaps/via/keymap.c (hunk 6 above)
Comment (sergiopoverony) [reply keyboards/sergiopoverony/creator_pro/keymaps/via/keymap.c]: enumed

**Review by drashna — APPROVED** [review]

**Comment by sergiopoverony** [comment]
Please pull to Master

File: keyboards/sergiopoverony/creator_pro/readme.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  * Chassis is PCB with supports
+  * PCB fabrication
+  * Uses a Pro Micro wint MicroUSB connector
+  * Includes a reset switch accessible by a hole on the bottom
+
+Make example for this keyboard (after setting up your build environment):
+
+    make sergiopoverony/creator_pro:default
+	or
+	make sergiopoverony/creator_pro:via
```
Comment (fauxpark) [inline keyboards/sergiopoverony/creator_pro/readme.md:27]:
```suggestion
```

File: keyboards/sergiopoverony/creator_pro/rules.mk [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = caterina
+
+MOUSEKEY_ENABLE = yes
+EXTRAKEY_ENABLE = yes
+CONSOLE_ENABLE  = yes
+COMMAND_ENABLE  = yes
+ENCODER_ENABLE  = yes
```
Comment (fauxpark) [inline keyboards/sergiopoverony/creator_pro/rules.mk:10]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sergiopoverony/creator_pro/creator_pro.c [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+}
+void turn_off_leds(void)
+{
+    writePinLow(RED_LED);
+    writePinLow(BLUE_LED);
+    writePinLow(GREEN_LED);
+}
+void turn_on_led(pin_t pin)
+{
+    writePinHigh(pin);
```
Comment (drashna) [inline keyboards/sergiopoverony/creator_pro/creator_pro.c:35]:
```suggestion
void matrix_init_kb(void) {
    matrix_init_user();
/* led pins */
    setPinOutput(RED_LED);
    setPinOutput(BLUE_LED);
    setPinOutput(GREEN_LED);
}
void turn_off_leds(void) {
    writePinLow(RED_LED);
    writePinLow(BLUE_LED);
    writePinLow(GREEN_LED);
}
void turn_on_led(pin_t pin) {
    writePinHigh(pin);
```

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**